File: WebSite with CosmosDB/server/Basketball/Microservice/ViewModels/ParticipatingPlayerVM.cs
using NBA.Infrastructure;
using NBA.Models;

namespace NBA.ViewModels
{
    public class ParticipatingPlayerVM : BasketballViewModel
    {
        public string PlayerId { get; set; }
        public string PlayerName { get; set; }
        public string ParticipationId { get; set; }
        public string TeamId { get; set; }
        public string TeamName { get; set; }
        public List<GamePlay> Plays { get; set; } = [];

        public override (bool Success, string Message) Validate()
        {
            return (false, "this viewModel cannot be used for write operations");
        }

        internal static ParticipatingPlayerVM FactorFrom(Participation model)
        {
            return new()
            {
                PlayerId = model.PlayerId,
                PlayerName = model.PlayerName,
                ParticipationId = model.Id,
                TeamId = model.TeamId,
                TeamName = model.TeamName,
                Plays = model.Plays
            };
        }
    }
}

File: Vue-Nuxt with Java-MongoDB/server/Basketball/Microservice/src/main/java/com/nba/microservice/viewmodels/ParticipatingPlayerVM.java
package com.nba.microservice.viewmodels;

import com.nba.microservice.models.Participation;
import com.nba.microservice.models.ValueObjects.GamePlay;
import com.nba.microservice.infrastructure.BasketballViewModel;
import com.nba.microservice.infrastructure.ValidationResult;
import java.util.ArrayList;
import java.util.List;

public class ParticipatingPlayerVM extends BasketballViewModel {

    private String playerId;
    private String playerName;
    private String participationId;
    private String teamId;
    private String teamName;
    private List<GamePlayVM> plays;

    public String getPlayerId() {
        return playerId;
    }

    public void setPlayerId(String playerId) {
        this.playerId = playerId;
    }

    public String getPlayerName() {
        return playerName;
    }

    public void setPlayerName(String playerName) {
        this.playerName = playerName;
    }

    public String getParticipationId() {
        return participationId;
    }

    public void setParticipationId(String participationId) {
        this.participationId = participationId;
    }

    public String getTeamId() {
        return teamId;
    }

    public void setTeamId(String teamId) {
        this.teamId = teamId;
    }

    public String getTeamName() {
        return teamName;
    }

    public void setTeamName(String teamName) {
        this.teamName = teamName;
    }

    public List<GamePlayVM> getPlays() {
        return plays;
    }

    public void setPlays(List<GamePlayVM> plays) {
        this.plays = plays;
    }

    @Override
    public ValidationResult validate() {
        return new ValidationResult(false, "this viewModel cannot be used for write operations");
    }

    public static ParticipatingPlayerVM factorFrom(Participation model) {
        ParticipatingPlayerVM playerVM = new ParticipatingPlayerVM();
        playerVM.setPlayerId(model.getPlayerId());
        playerVM.setPlayerName(model.getPlayerName());
        playerVM.setParticipationId(model.getId());
        playerVM.setTeamId(model.getTeamId());
        playerVM.setTeamName(model.getTeamName());

        List<GamePlayVM> gamePlayVMs = new ArrayList<>();
        for (GamePlay play : model.getPlays()) {
            gamePlayVMs.add(GamePlayVM.factoryFrom(play));
        }
        playerVM.setPlays(gamePlayVMs);

        return playerVM;
    }
}